File: katas/7 kyu/growth-of-a-population/solution.py
import codewars_test as test
# -------------------------------- my solution ------------------------------- #
def nb_year(p0, percent, aug, p):
  years = 0
  while p0 <= p :
    years += 1
    p0 += int(p0 * percent/100) + aug
  return years
# ------------------------------ clever solution ----------------------------- #
def nb_year(p0, percent, aug, p, years = 0):
    if p0 < p:
        return nb_year(p0 + int(p0 * percent / 100) + aug, percent, aug, p, years + 1)
    return years
# ----------------------------------- test ----------------------------------- #
@test.describe("nb_year")
def tests():
    @test.it("basic tests")
    def basics():
        test.assert_equals(nb_year(1500, 5, 100, 5000), 15)
        test.assert_equals(nb_year(1500000, 2.5, 10000, 2000000), 10)
        test.assert_equals(nb_year(1500000, 0.25, 1000, 2000000), 94)

File: katas/7 kyu/growth-of-a-population/solution.js
// growth-of-a-population
/* ------------------------------- MY SOLUTION ------------------------------ */
function nbYear(p0, percent, aug, p) {
  years = 0
  while (p0 < p) {
    years += 1
    p0 += parseInt(p0 * percent/100) + aug // parseInt or Math.floor
  }
  return years
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
const chai = require("chai");
const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
chai.config.truncateThreshold=0;

describe("nbYear",function() {
  it("Basic tests",function() {    
      Test(nbYear(1500, 5, 100, 5000), 15);
      Test(nbYear(1500000, 2.5, 10000, 2000000), 10);
      Test(nbYear(1500000, 0.25, 1000, 2000000), 94);
  })})